### PR TITLE
VAR-429, VAR-432 | Implement sorting resources by accessibility

### PR DESCRIPTION
**Pull request by ahie**
## Description :sparkles:

Adds sorting by accessibility to the resource search page.

This commit additionally connects the selected accessibility preferences
to the redux controlled state. This enabled to e.g. update the
resource listing's accessibility icons instantly as preferences are
changed.

### Closes :no_good_woman:

[VAR-429](https://helsinkisolutionoffice.atlassian.net/browse/VAR-429)
[VAR-432](https://helsinkisolutionoffice.atlassian.net/browse/VAR-432)

### Automated tests :gear:️

`app/state/reducers/__tests__/accessibilityReducer.test.js `

### Manual testing :construction_worker_man:

- On the front page, search for resources.
- Once the results are listed, toggle sorting by accessibility from the dropdown menu.
- The resources should be refetched, with resources that have shortcomings for your accessibility preferences listed last.

### CODE REVIEW

File: src/domain/resource/card/ResourceCard.js [hunk 1]
```diff
@@ -187,9 +189,14 @@ class ResourceCard extends React.Component {
...
-                  nShortcomings: accessibilityShortcomingsCount,
-                })
+                ? t(
+                  accessibilityShortcomingsCount === 1
+                    ? 'ResourceInfo.foundAccessibilityShortcoming'
+                    : 'ResourceInfo.foundAccessibilityShortcomings',
+                  {
+                    nShortcomings: accessibilityShortcomingsCount,
+                  },
+                )
```
Comment (Nizar-Rahme) [inline src/domain/resource/card/ResourceCard.js:199]: Wondering whether `react-intl` has an official way to handle pluralization.

File: app/state/reducers/ui/accessibilityReducer.js [hunk 2]
```diff
@@ -17,6 +18,12 @@ function toggleContrast(state) {
   return { ...state, isHighContrast: !state.isHighContrast };
 }
 
+function setAccessibilityPreferences(state, preferences) {
+  const newPreferences = Array.isArray(preferences) ? preferences : [];
+  setSelA11yPref(newPreferences);
```
Comment (Nizar-Rahme) [inline app/state/reducers/ui/accessibilityReducer.js:23]: I think should be pure functions and we should avoid performing side effects inside them [according to Redux docs](https://redux.js.org/basics/reducers#handling-actions).
Probably the subscriber should handle persisting and retrieving the state from the `localStorage`(https://stackoverflow.com/a/35675304/9441362)

**Review by Nizar-Rahme — CHANGES_REQUESTED** [review]

File: src/domain/resource/card/ResourceCard.js (hunk 1 above)
Comment (ahie) [reply src/domain/resource/card/ResourceCard.js:199]: There are components, but I don't think there's a simpler way with the `t` function. In any case you need to provide the translations manually for singular and plural forms.

File: app/state/reducers/ui/accessibilityReducer.js (hunk 2 above)
Comment (ahie) [reply app/state/reducers/ui/accessibilityReducer.js]: This is true, moved it to a separate middleware.

**Review by Nizar-Rahme — APPROVED** [review]
LGTM 👍

**Pull request merged**